backend/api/views.py

```
@@ -1,75 +1,163 @@
+        serializer = self.get_serializer(instance, data=request.data, partial=True)
         if serializer.is_valid():
             serializer.validated_data['onboarding_completed'] = True
             serializer.save()
             return Response(serializer.data, status=status.HTTP_200_OK)
-        else:
-            return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
 
-class SetFirebaseTokenView(APIView):
+class SetFirebaseTokenView(generics.CreateAPIView):
     permission_classes = [AllowAny]
```

Ruff (0.8.2)

Mutable class attributes should be annotated with `typing.ClassVar`

(RUF012)